### PR TITLE
check for valid vfu_setup_region() flags

### PR DESCRIPTION
**Pull request by jlevon**

### CODE REVIEW

**Comment by jlevon** [comment]
I'll wait on Jag's PR

File: lib/libvfio-user.c [hunk 1]
```diff
@@ -1430,6 +1430,17 @@ vfu_setup_region(vfu_ctx_t *vfu_ctx, int region_idx, size_t size,
 
     assert(vfu_ctx != NULL);
 
+    if ((flags & ~(VFU_REGION_FLAG_MASK)) ||
+        (!(flags & VFU_REGION_FLAG_RW))) {
```
Comment (tmakatos) [inline lib/libvfio-user.c:1434]: Probably not important, but why not allow a region is not readable/writeable, is not an I/O region, and there are no callabacks?

**Review by tmakatos — APPROVED** [review]

File: lib/libvfio-user.c (hunk 1 above)
Comment (jlevon) [reply lib/libvfio-user.c:1434]: What would the point in such a region be? nothing could ever interact with it could it?

Comment (tmakatos) [reply lib/libvfio-user.c:1434]: agreed

**Pull request merged**